src/build.py

```diff
@@ -604,49 +611,50 @@ def NoSync(*args):
   pass
 
 
-ALL_SOURCES = [
-    Source('waterfall', SCRIPT_DIR, None, custom_sync=NoSync),
-    Source('llvm', LLVM_PROJECT_SRC_DIR,
-           LLVM_GIT_BASE + 'llvm-project.git'),
-    Source('llvm-test-suite', LLVM_TEST_SUITE_SRC_DIR,
-           LLVM_MIRROR_BASE + 'test-suite.git'),
-    Source('emscripten', EMSCRIPTEN_SRC_DIR,
-           EMSCRIPTEN_GIT_BASE + 'emscripten.git',
-           checkout=RemoteBranch('incoming')),
-    Source('fastcomp', FASTCOMP_SRC_DIR,
-           EMSCRIPTEN_GIT_BASE + 'emscripten-fastcomp.git',
-           checkout=RemoteBranch('incoming')),
-    Source('fastcomp-clang',
-           os.path.join(FASTCOMP_SRC_DIR, 'tools', 'clang'),
-           EMSCRIPTEN_GIT_BASE + 'emscripten-fastcomp-clang.git',
-           checkout=RemoteBranch('incoming')),
-    Source('gcc', GCC_SRC_DIR,
-           GIT_MIRROR_BASE + 'chromiumos/third_party/gcc.git',
-           checkout=GCC_REVISION, depth=GCC_CLONE_DEPTH),
-    Source('v8', V8_SRC_DIR,
-           GIT_MIRROR_BASE + 'v8/v8.git',
-           custom_sync=ChromiumFetchSync),
-    Source('host-toolchain', PREBUILT_CLANG,
-           GIT_MIRROR_BASE + 'chromium/src/tools/clang.git',
-           custom_sync=SyncToolchain),
-    Source('cr-buildtools', os.path.join(WORK_DIR, 'build'),
-           GIT_MIRROR_BASE + 'chromium/src/build.git'),
-    Source('cmake', '', '',  # The source and git args are ignored.
-           custom_sync=SyncPrebuiltCMake),
-    Source('nodejs', '', '',  # The source and git args are ignored.
-           custom_sync=SyncPrebuiltNodeJS),
-    Source('gnuwin32', '', '',  # The source and git args are ignored.
-           custom_sync=SyncGNUWin32),
-    Source('wabt', WABT_SRC_DIR,
-           WASM_GIT_BASE + 'wabt.git'),
-    Source('binaryen', BINARYEN_SRC_DIR,
-           WASM_GIT_BASE + 'binaryen.git'),
-    Source('musl', MUSL_SRC_DIR,
-           MUSL_GIT_BASE + 'musl.git',
-           checkout=RemoteBranch('wasm-prototype-1')),
-    Source('java', '', '',  # The source and git args are ignored.
-           custom_sync=SyncPrebuiltJava)
-]
+def AllSources():
```

Comment:
Why change this to a function?

Reply:
The hardcoded paths above are switched to functions so they can be evaluated after the command-line flags are parsed (since the flags can now change the working directories). Likewise with `AllSources`; since it uses the source directories, it has to be called after flag parsing rather than on startup.